### PR TITLE
Add django base web application

### PR DESCRIPTION
**Pull request by La0**
Add an empty django web application, shipped through CI/CD

### CODE REVIEW

File: backend/code_review_backend/app/settings.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    if "DATABASE_URL" in os.environ:
+        logger.info("Using remote database from $DATABASE_URL")
+        DATABASES["default"] = dj_database_url.parse(
+            os.environ["DATABASE_URL"], ssl_require=True
+        )
+    else:
+        logger.info("DATABASE_URL not found, will use sqlite. Data may be lost.")
+
+    # Insert Whitenoise Middleware after the security one
+    MIDDLEWARE.insert(1, "whitenoise.middleware.WhiteNoiseMiddleware")
```
Comment (La0) [inline backend/code_review_backend/app/settings.py]: Added these lines to run on Heroku

File: backend/code_review_backend/app/settings.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+REST_FRAMEWORK = {
+    # Use Django's standard `django.contrib.auth` permissions,
+    # or allow read-only access for unauthenticated users.
+    "DEFAULT_PERMISSION_CLASSES": [
+        "rest_framework.permissions.IsAuthenticatedOrReadOnly"
+    ],
+    # Setup pagination
+    "PAGE_SIZE": 50,
+    "DEFAULT_PAGINATION_CLASS": "rest_framework.pagination.LimitOffsetPagination",
+}
```
Comment (La0) [inline backend/code_review_backend/app/settings.py]: Config for the API (could be in #162)

File: backend/code_review_backend/app/settings.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+# Application definition
+
+INSTALLED_APPS = [
+    "django.contrib.admin",
+    "django.contrib.auth",
+    "django.contrib.contenttypes",
+    "django.contrib.sessions",
+    "django.contrib.messages",
+    "django.contrib.staticfiles",
+    "rest_framework",
```
Comment (La0) [inline backend/code_review_backend/app/settings.py]: Added that line to load DRF

File: backend/code_review_backend/app/settings.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/2.2/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = "t!+s!@x5p!85x19q83jufr#95_z0fv7$!u5z*c&gi!%hr3^w+r"
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
```
Comment (marco-c) [inline backend/code_review_backend/app/settings.py]: Can you add a comment here to explain that, for production, this is overwritten in the Heroku section at the end of the file?

Comment (La0) [reply backend/code_review_backend/app/settings.py]: Done

File: backend/code_review_backend/app/settings.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+ROOT_DIR = os.path.dirname(BASE_DIR)
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/2.2/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = os.environ.get(
+    "SECRET_KEY", "t!+s!@x5p!85x19q83jufr#95_z0fv7$!u5z*c&gi!%hr3^w+r"
+)
```
Comment (marco-c) [inline backend/code_review_backend/app/settings.py]: Maybe you could move the env part to the Heroku section too, so you have the same behavior as for DEBUG

**Pull request merged**